io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/DuplicateIDsFilterIterator.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2024 wetransform GmbH
+ * 
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution. If not, see <http://www.gnu.org/licenses/>.
+ * 
+ * Contributors:
+ *     wetransform GmbH <http://www.wetransform.to>
+ */
+
+package eu.esdihumboldt.hale.io.gml.reader.internal.wfs;
+
+import java.util.HashSet;
+import java.util.NoSuchElementException;
+
+import javax.xml.namespace.QName;
+
+import eu.esdihumboldt.hale.common.instance.model.Instance;
+import eu.esdihumboldt.hale.common.instance.model.ext.InstanceIterator;
+import eu.esdihumboldt.hale.common.schema.model.TypeDefinition;
+import eu.esdihumboldt.hale.io.gml.geometry.GMLConstants;
+
+/**
+ * Filter the instances by gml:id, storing only those with unique IDs.
+ */
+public class DuplicateIDsFilterIterator implements InstanceIterator {
+
+	private final HashSet<String> uniqueIDInstances = new HashSet<String>();
+	private final InstanceIterator iterator;
+	private Instance nextInstance;
+	private boolean instranceAlreadyReturned = false;
```

Comment:
I think it would help documenting for the variable how it is used and what it represents (so basically that it tells us if `next()` or `skip()` have already been called for the current instance in `nextInstance`).